### ISSUE TITLE
Failed to register direct-download template in 4.18.1.0-RC1

### ISSUE THREAD

**Issue by weizhouapache** [apache/cloudstack#7929]
![image](https://github.com/apache/cloudstack/assets/57355700/e57b7413-e889-4266-b022-93c91c6ba0a8)

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~

~~~

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by weizhouapache** [comment]
the error
```
2023-08-31 08:00:50,678 DEBUG [c.c.a.t.Request] (AgentManager-Handler-10:null) (logid:) Seq 2-4678958537861260053: Processing:  { Ans: , MgmtId: 32986774897485, via: 2, Ver: v1, Flags: 10, [{"org.apache.cloudstack.agent.directdownload.CheckUrlAnswer":{"templateSize":"(-0.00 KB) -1","result":"true","wait":"0","bypassHostMaintenance":"false"}}] }
```

it might be because the direct-download certificates are not uploaded, but we need to fix the error anyway

**Comment by weizhouapache** [comment]
the issue still exist after uploading directdownload template certificates
this may be related to #7693 and #7923

**Comment by weizhouapache** [comment]
@nvazquez  
can you please have a look ? thanks

**Comment by weizhouapache** [comment]
it works fine in 4.17.2.0

- before uploading direct-download template certificates

![image](https://github.com/apache/cloudstack/assets/57355700/6eab0d24-8455-4630-aa6d-809026ee557d)

- after uploading direct-download template certificates
![image](https://github.com/apache/cloudstack/assets/57355700/b2421cb9-8065-42b6-b2ec-1e9e8ca6af87)

**Comment by weizhouapache** [comment]
for your information,

- the template url is https://download.cloudstack.org/systemvm/4.18/systemvmtemplate-4.18.1-kvm.qcow2.bz2

- the certificate is retrieved by
`openssl s_client -showcerts -servername download.cloudstack.org -connect download.cloudstack.org:443 </dev/null`

- the certificate is saved as `cert` file on local system

```
cat > cert << "EOF"
-----BEGIN CERTIFICATE-----
MIIF/DCCBOSgAwIBAgISA0unS6sIh08f4WYmmSucrLozMA0GCSqGSIb3DQEBCwUA
MDIxCzAJBgNVBAYTAlVTMRYwFAYDVQQKEw1MZXQncyBFbmNyeXB0MQswCQYDVQQD
EwJSMzAeFw0yMzA4MjcwMzI1MTFaFw0yMzExMjUwMzI1MTBaMCIxIDAeBgNVBAMT
F2Rvd25sb2FkLmNsb3Vkc3RhY2sub3JnMIICIjANBgkqhkiG9w0BAQEFAAOCAg8A
MIICCgKCAgEA3+cPWpDfkUudN6AL36cxs10skSIln7chag+dRKZXZU7F7RvdV7Fa
mc0avLh5CE8HlVuixqj7jC5of1eu4ri6PijCJt/3NaHQZsLdp6Ydw63LWRl7dFaf
AgWi35M6YX8qffi7AFDXphuwzU2zxe9Wp6GlS+BBBDvXlB4DuOP86yvFDCSLitqa
cun+/hyeXEZtsyzzC1kI4q5kdfiCi+LmfCbnbtOmo4hYS6ITf0isW1WLp3d6hLcg
bC9sNXJ/mjMA3cy0ErOfppUEHaUNfjQHAiMf3F9+PVkJyLVNUwLPxEMZgOTaz0Yf
afuAzHyJzLrO1pLejd51fGB04qBpBkTerjx6t3tpvplJaAiaBl8OBfZV0bJEXwzp
qepBdQxxwKZmLHaE5jnVKYFclku/3ueQ67opbajpaKbA3RXaJ3fIFMF4vuxb6NBC
ny+NQmonNi3U/Rwub8Ghip9kO4U+U9z/wLGJvdzNsvoWUwRimQq1mB0zGG0uvJ/Y
KNh50YfEV1qd5jbjQ8hA5IwQbIfRmf5IbZ6wQzWjUFP9juPKs7eRd+APc+XAsHQW
XeVUwcby2UoI4Zmip6JlUIaQMgjiViA4+kZ03+JSGvFDfjuy5scAL2gLJQ0o60Hd
1osaDkZviLtpVyf/IIjt/oznzL7L6LhicKD+C+X5liPf1/YMDcM3lHcCAwEAAaOC
AhowggIWMA4GA1UdDwEB/wQEAwIFoDAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYB
BQUHAwIwDAYDVR0TAQH/BAIwADAdBgNVHQ4EFgQUl7fTXI7dKw7EbVT2gk9KK7si
8lcwHwYDVR0jBBgwFoAUFC6zF7dYVsuuUAlA5h+vnYsUwsYwVQYIKwYBBQUHAQEE
STBHMCEGCCsGAQUFBzABhhVodHRwOi8vcjMuby5sZW5jci5vcmcwIgYIKwYBBQUH
MAKGFmh0dHA6Ly9yMy5pLmxlbmNyLm9yZy8wIgYDVR0RBBswGYIXZG93bmxvYWQu
Y2xvdWRzdGFjay5vcmcwEwYDVR0gBAwwCjAIBgZngQwBAgEwggEFBgorBgEEAdZ5
AgQCBIH2BIHzAPEAdgB6MoxU2LcttiDqOOBSHumEFnAyE4VNO9IrwTpXo1LrUgAA
AYo1PRrKAAAEAwBHMEUCIQD22aIgUaZOOkE/4z9tqMuhDnoPrtBMfgp+TBvNYSuw
igIgUBPywebeAFnWjBYDI6oMse7qnpgopCB9Cer+WJnaNJYAdwDoPtDaPvUGNTLn
Vyi8iWvJA9PL0RFr7Otp4Xd9bQa9bgAAAYo1PRrJAAAEAwBIMEYCIQC+4NNCNCIU
1ltKydMffyGClNJ3Aa4bPWQNwCaPV6liWgIhAPM3wtltl79Ry6uu/keIfHRFTULJ
Qtx1qDkWWGQTinxoMA0GCSqGSIb3DQEBCwUAA4IBAQBK49Y/eK2TdQyyHlcCVOMG
g2IaWeu5Ba4zNQDj3HhRdm66B1mA+5eVYS/oxcvpN+zAmMvNRJZRyq+2bNZeUDTq
T9ztoSR2XRppY/0W6xVzcFatcb6wWHD+YIdY7AxNRhmgRs2q2TlR/V9b1Qpial18
OU5LsGryIGHZX0g0QS/wiXGyv/xyupJR55kGQUKwBNKg9Z0UVFqEPjf9znadS9Xh
ZMjCK+WF2fh1+/+pNy0s2bLJGDNiAifnXS0sswk12GlinjBStJ5znoClVuyKE2uc
hOCEFErZI/O80qSRuP7WzpsRpu8CAFsHvCSeDN1msBmF1XCh79XZlK0cjn/f8Jf0
-----END CERTIFICATE-----
-----BEGIN CERTIFICATE-----
MIIFFjCCAv6gAwIBAgIRAJErCErPDBinU/bWLiWnX1owDQYJKoZIhvcNAQELBQAw
TzELMAkGA1UEBhMCVVMxKTAnBgNVBAoTIEludGVybmV0IFNlY3VyaXR5IFJlc2Vh
cmNoIEdyb3VwMRUwEwYDVQQDEwxJU1JHIFJvb3QgWDEwHhcNMjAwOTA0MDAwMDAw
WhcNMjUwOTE1MTYwMDAwWjAyMQswCQYDVQQGEwJVUzEWMBQGA1UEChMNTGV0J3Mg
RW5jcnlwdDELMAkGA1UEAxMCUjMwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEK
AoIBAQC7AhUozPaglNMPEuyNVZLD+ILxmaZ6QoinXSaqtSu5xUyxr45r+XXIo9cP
R5QUVTVXjJ6oojkZ9YI8QqlObvU7wy7bjcCwXPNZOOftz2nwWgsbvsCUJCWH+jdx
sxPnHKzhm+/b5DtFUkWWqcFTzjTIUu61ru2P3mBw4qVUq7ZtDpelQDRrK9O8Zutm
NHz6a4uPVymZ+DAXXbpyb/uBxa3Shlg9F8fnCbvxK/eG3MHacV3URuPMrSXBiLxg
Z3Vms/EY96Jc5lP/Ooi2R6X/ExjqmAl3P51T+c8B5fWmcBcUr2Ok/5mzk53cU6cG
/kiFHaFpriV1uxPMUgP17VGhi9sVAgMBAAGjggEIMIIBBDAOBgNVHQ8BAf8EBAMC
AYYwHQYDVR0lBBYwFAYIKwYBBQUHAwIGCCsGAQUFBwMBMBIGA1UdEwEB/wQIMAYB
Af8CAQAwHQYDVR0OBBYEFBQusxe3WFbLrlAJQOYfr52LFMLGMB8GA1UdIwQYMBaA
FHm0WeZ7tuXkAXOACIjIGlj26ZtuMDIGCCsGAQUFBwEBBCYwJDAiBggrBgEFBQcw
AoYWaHR0cDovL3gxLmkubGVuY3Iub3JnLzAnBgNVHR8EIDAeMBygGqAYhhZodHRw
Oi8veDEuYy5sZW5jci5vcmcvMCIGA1UdIAQbMBkwCAYGZ4EMAQIBMA0GCysGAQQB
gt8TAQEBMA0GCSqGSIb3DQEBCwUAA4ICAQCFyk5HPqP3hUSFvNVneLKYY611TR6W
PTNlclQtgaDqw+34IL9fzLdwALduO/ZelN7kIJ+m74uyA+eitRY8kc607TkC53wl
ikfmZW4/RvTZ8M6UK+5UzhK8jCdLuMGYL6KvzXGRSgi3yLgjewQtCPkIVz6D2QQz
CkcheAmCJ8MqyJu5zlzyZMjAvnnAT45tRAxekrsu94sQ4egdRCnbWSDtY7kh+BIm
lJNXoB1lBMEKIq4QDUOXoRgffuDghje1WrG9ML+Hbisq/yFOGwXD9RiX8F6sw6W4
avAuvDszue5L3sz85K+EC4Y/wFVDNvZo4TYXao6Z0f+lQKc0t8DQYzk1OXVu8rp2
yJMC6alLbBfODALZvYH7n7do1AZls4I9d1P4jnkDrQoxB3UqQ9hVl3LEKQ73xF1O
yK5GhDDX8oVfGKF5u+decIsH4YaTw7mP3GFxJSqv3+0lUFJoi5Lc5da149p90Ids
hCExroL1+7mryIkXPeFM5TgO9r0rvZaBFOvV2z0gp35Z0+L4WPlbuEjN/lxPFin+
HlUjr8gRsI3qfJOQFy/9rKIJR0Y/8Omwt/8oTWgy1mdeHmmjk7j1nYsvC9JSQ6Zv
MldlTTKB3zhThV1+XWYp6rjd5JW1zbVWEkLNxE7GJThEUG3szgBVGP7pSWTUTsqX
nLRbwHOoq7hHwg==
-----END CERTIFICATE-----
EOF
```

- the certificate is uploaded on management server by cmk
```
cmk upload templatedirectdownloadcertificate hypervisor=kvm name=download.cloudstack.org certificate="$(cat cert)" zoneid=$(cmk list zones |jq -r '.zone[].id' |head -n1)
```

**Comment by weizhouapache** [comment]
update: it seems to be because the image is compressed by bzip2
this does not work: https://download.cloudstack.org/systemvm/4.18/systemvmtemplate-4.18.1-kvm.qcow2.bz2
this works: https://download.cloudstack.org/templates/capi-test/kvm/ubuntu-2004-kube-v1.23.2-kvm.qcow2

**Comment by weizhouapache** [comment]
fixed by #7932 
thanks @nvazquez
